src/test/kotlin/se/uulm/snowballr/backend/repository/CriterionTableRepoTest.kt

```
@@ -135,7 +137,7 @@ class CriterionTableRepoTest : RepositoryTest(arrayOf(CriterionTable, ProjectTab
                         .setProjectId(UUID.randomUUID().toString())
                         .build()
 
-                assertThrows<NotFoundException> { repo.createCriterion(request, testUserId) }
+                assertThrows<SQLException> { repo.createCriterion(request, testUserId) }
```

I think the problem now is, that we throw a SQLException which is not encapsulated as Result type, i.e. not checked in the service layer but only thrown. However, our validation layer does not catch this exception as well and we wouldn't even know the reason for the SQLException, so this seems not the best solution to me. Either we should use the Result type as well or we should still use this `getEntityById` helpers.

---

Actually this isn't the biggest problem for now, as we actually always check access rights and whether something exists etc. before we retrieve something from the repo layer, but I am not sure whether this is the most elegant way.

---

I agree. All of these tests where a `SQLException` is thrown, we actually test whether our DB design prohibits invalid actions. Do you have an idea how we should move forward?

---

Yeah, actually this isn't the worst idea as we then test whether our constraints are ok. So I think we can keep these tests, however, maybe it could be advantageous to catch such errors in the exception interceptor and log a warning the backend or so in case it happens so we notice that there could be a bug in the backend? Otherwise I would just ignore them for now and think about these tests when we refactor the access rights.

---

We don't ignore any exception. Every unexpected exception is handled in the ExceptionInterceptor and mapped to an `INTERNAL` status code while the exception is logged.

---

Ah ok, perfect.